zaino-state/src/local_cache.rs

```diff
@@ -51,6 +58,7 @@ impl BlockCache {
 
         Ok(BlockCache {
             fetcher: fetcher.clone(),
+            state: state.map(|state| state.clone()),
```

Comment:
Non-blocking, `Option::cloned` is a helper for this. 
```suggestion
            state: state.cloned(),
```

Reply:
Oh nice! Added in [53f1839](https://github.com/zingolabs/zaino/pull/264/commits/53f1839e30d23173a41167039f69b967830763f1)